bastion/src/supervisor.rs

```diff
@@ -1689,12 +1665,108 @@ impl Supervised {
     }
 }
 
+impl RestartStrategy {
+    /// Creates a new instance of RestartStrategy.
+    ///
+    /// # Arguments
+    ///
+    /// * `max_retries` - Defines an amout of attempts to restore an actor
+    /// to the initial state. Set `None` to have unlimited amout of attemps.
+    ///
+    /// * `strategy` - The strategy to use:
+    ///     - [`ActorRestartStrategy::Immediate`] would restart the
+    ///         failed actor as soon as possible.
+    ///     - [`ActorRestartStrategy::LinearBackOff`] would restart the
+    ///         failed actor with the delay increasing linearly.
+    ///     - [`ActorRestartStrategy::ExponentialBackOff`] would restart the
+    ///         failed actor with the delay, multiplied by given coefficient.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use std::time::Duration;
+    /// # use bastion::prelude::*;
+    /// #
+    /// # fn main() {
+    /// #     use std::time::Duration;
+    /// #     let actor_restart_strategy = ActorRestartStrategy::LinearBackOff {
+    /// #         timeout: Duration::from_secs(5)
+    /// #     };
+    /// #     let restart_strategy = RestartStrategy::default()
+    /// #        .with_actor_restart_strategy(actor_restart_strategy);
```

Comment:
This just tests the initiation, that's nice but we need an actual test for this case in `tests/` I think:
https://github.com/bastion-rs/bastion/tree/master/bastion/tests

Reply:
Done.